From 8cfa786dfbf369b3d4086c0a11d1e1c9c176118a Mon Sep 17 00:00:00 2001
From: Xingang Wang <wangxingang5@huawei.com>
Date: Sat, 29 Jan 2022 15:10:45 +0800
Subject: [PATCH] dt-bindings: mpam: refactor device tree node structure
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 8cfa786dfbf369b3d4086c0a11d1e1c9c176118a
Modified-by-SEL: No


arm64/mpam: refactor device tree structure to support multiple
devices

ascend inclusion
category: feature
bugzilla:
https://gitee.com/openeuler/kernel/issues/I49RB2
CVE: NA

---------------------------------------------------

To support multiple mpam device nodes, all nodes should be organized
as child of the same parent nodes. This makes sure that the mpam
discovery start and complete procedure in the right execution order.
Add modification in the devicetree documentation to record this.

Signed-off-by: Xingang Wang <wangxingang5@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 .../devicetree/bindings/arm/arm,mpam.txt      | 45 ++++++++++---------
 1 file changed, 24 insertions(+), 21 deletions(-)

diff --git a/Documentation/devicetree/bindings/arm/arm,mpam.txt b/Documentation/devicetree/bindings/arm/arm,mpam.txt
index 65c1e6809685..e9ba09bb3159 100644
--- a/Documentation/devicetree/bindings/arm/arm,mpam.txt
+++ b/Documentation/devicetree/bindings/arm/arm,mpam.txt
@@ -28,27 +28,30 @@ and Monitoring (MPAM), for Armv8-A", MPAM interrupts(section 8.8).
 
 Example:
 
-mpam_memory0 {
+mpam {
 	compatible = "arm,mpam";
-	reg = <0x0 0x10000000 0x0 0x10000>;
-	type = <2>; /* memory type */
-	numa-node-id = <0>;
-	overflow-interrupt = <0>;
-	overflow-flags = <0>;
-	error-interrupt = <0>;
-	error-interrupt-flags = <0>;
-	not-ready-max = <0>;
-};
 
-mpam_cache0 {
-	compatible = "arm,mpam";
-	reg = <0x0 0x20000000 0x0 0x10000>;
-	type = <1>; /* cache type */
-	cache-id = <0>;
-	cache-level = <3>;
-	overflow-interrupt = <0>;
-	overflow-flags = <0>;
-	error-interrupt = <0>;
-	error-interrupt-flags = <0>;
-	not-ready-max = <0>;
+	mpam_memory0 {
+		reg = <0x0 0x10000000 0x0 0x10000>;
+		type = <2>; /* memory type */
+		numa-node-id = <0>;
+		overflow-interrupt = <0>;
+		overflow-flags = <0>;
+		error-interrupt = <0>;
+		error-interrupt-flags = <0>;
+		not-ready-max = <0>;
+	};
+
+	mpam_cache0 {
+		reg = <0x0 0x20000000 0x0 0x10000>;
+		type = <1>; /* cache type */
+		cache-id = <0>;
+		cache-level = <3>;
+		overflow-interrupt = <0>;
+		overflow-flags = <0>;
+		error-interrupt = <0>;
+		error-interrupt-flags = <0>;
+		not-ready-max = <0>;
+	};
+
 };
-- 
2.26.2

